### PR TITLE
Adds all sharp items that have a force >=10 to the list of items that can be used for the saw bone step in ghetto surgery

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR adds all sharp items that have a force >=10 to the list of items that can attempt to perform the "saw bone" stock surgery step, and gives them an implement value of 20 (not necessarily a *success rate* of 20%, because the recent changes to surgery failure chance have made actually determining your likelyhood to fail a step when performing ghetto surgery kind of complex).

## Why It's Good For The Game

The saw bone surgery step is the ONLY stock surgery step that cannot be completed with common items. Currently, the only items that can perform ghetto saw bone surgery are: arm blades, fire axes, hatchets, and butcher knives, and all of those are either antag items or specific to one or two departments. It should be noted that you can also perform the saw surgery with a tool value of 100 if you use a chainsaw or certain null rod forms, but again, not everyone has access to those.

Since all of the other surgery steps can be completed with much more common items (the repair body step of the tend wounds surgery can be completed with a PEN, and you can clamp bleeders with CABLE COILS) and the saw bone surgery step is required for a lot of fun surgeries (such as organ manipulation for ash walkers, so they can finally implant legion cores in each other), I've made the saw bone surgery completable with items that are more accessible to the common man/crew member (and items that could logically cut bone, such as eswords). By the way, in case you weren't already aware, the incise step already can be performed with any sharp item, so this kind of thing has precedent.

Hopefully, this PR should encourage more people to try ghetto surgery gimmicks/stunts, and ghetto surgery gimmicks are cool.

## Changelog
:cl: ATHATH
balance: All sharp items that have a force greater than or equal to 10 can be used to perform the saw bone surgery step with ghetto surgery, but they'll have a low success rate/step speed when doing so.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
hmmm today i will cut someone's arm off with a plastic knife

**Comment by ATH1909** [comment]
> hmmm today i will cut someone's arm off with a plastic knife

If you want me to, I could check for a force threshold.

**Comment by trollbreeder** [comment]
> If you want me to, I could check for a force threshold.

Please do.

**Comment by Fikou** [comment]
also hatchets can be gotten by literally any crewmember on most stations

**Comment by ATH1909** [comment]
Bam, done.

**Comment by ATH1909** [comment]
oh god oh fuck why is there a merge conflict

**Comment by Fikou** [comment]
hmmm today i will cut someone's arm off with gar glasses

**Comment by ATH1909** [comment]
> hmmm today i will cut someone's arm off with gar glasses

Eh, that's fine, gar glasses can crit someone in 10-11 hits anyway, so they're probably strong enough to cut through bone after sawing at it for a bit.

**Comment by trollbreeder** [comment]
hmmmmm
today i will remove the captain's liver with a ninja star

**Comment by itseasytosee** [comment]
Hmm, today I will cut bone with a scalpel?

**Comment by Fikou** [comment]
cut bone with a kitchen knife, i shall

**Comment by ATH1909** [comment]
> Hmm, today I will cut bone with a scalpel?

You'll have a high failure chance, but you can try. You can use the "wrong" implement (with a higher failure chance) for a couple of other surgery steps too.

**Comment by ATH1909** [comment]
> cut bone with a kitchen knife, i shall

I mean, you can turn a corpse into a pile of gibs with that knife, so...

Maybe I should make it check for the ability to butcher things instead of for force? Depends on what the maintainers want.

**Comment by trollbreeder** [comment]
Hmm, today i will perform coronary bypass with a fork.

**Comment by PKPenguin321** [comment]
i love ghetto analogues like this 👍

**Comment by HomocidlePancakes** [comment]
There are more sharp items with force >= 10 than <10. This is a bruh moment.

**Comment by trollbreeder** [comment]
Oh, oh, i have an idea!
What if the implement percent of a ghetto sharp item is equal to it's force?

**Comment by ATH1909** [comment]
> Oh, oh, i have an idea!
> What if the implement chance of a ghetto sharp item is equal to it's force?

I had that idea too, but the way that surgery code is set up would make that really hard to implement.



> There are more sharp items with force >= 10 than <10. This is a bruh moment.

I know. This is just to keep Fikou's immersion from being broken by allowing someone to saw through bone with a plastic knife.

**Comment by HomocidlePancakes** [comment]
Make chance scale with force or something.

**Comment by ExcessiveUseOfCobblestone** [comment]
just add the ghetto step, I don't understand the rationale behind the prob increase because you can still do these just fine.

**Comment by ATH1909** [comment]
> just add the ghetto step, I don't understand the rationale behind the prob increase because you can still do these just fine.

It's there so that I can give the new generic sharp item (with a force >=10) ghetto step a reasonable implement value without it being equivalent to/better than one of the existing ghetto steps' implement values.

**Comment by ExcessiveUseOfCobblestone** [comment]
1/4 (25) to 1/5 (20) seems fine with me. If anything make it lower than 20 rather than boost everything else up.

**Comment by ExcessiveUseOfCobblestone** [comment]
yes, just click resolve conflicts

**Comment by ATH1909** [comment]
> yes, just click resolve conflicts

I did that, but I don't know what to do from there.

**Comment by ExcessiveUseOfCobblestone** [comment]
on atom you can pick which one you want, otherwise you have to pick n choose the code you want (in this case it was just added code so you take everything from your end and delete the stuff between ===== and master) then delete the markup git has added.

**Comment by ExcessiveUseOfCobblestone** [comment]
I just resolved it for you

**Comment by ATH1909** [comment]
> I just resolved it for you

Thanks.

> just add the ghetto step, I don't understand the rationale behind the prob increase because you can still do these just fine.

I've also incorporated this suggestion as well; this PR now no longer modifies the success chances of the items that were previously capable of performing the saw bone step.

I'm gonna go test this PR now, and after that we should be good, yeah?

**Comment by ATH1909** [comment]
testingfailure

Uh, Houston, we have a problem. The PR... doesn't seem to have changed anything in my tests. I mostly copied the code over from the incise surgery step... are we sure that the incise step's generic sharp object option actually works? Like, has anyone tested the incise step with anything other than a scalpel, esword, knife, or glass shard? I'll try to do test the ghetto version of the incise step in a real round later today.

File: code/modules/surgery/organic_steps.dm [hunk 1]
```diff
@@ -103,15 +103,20 @@
...
 	time = 54
 
 /datum/surgery_step/saw/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	display_results(user, target, "<span class='notice'>You begin to saw through the bone in [target]'s [parse_zone(target_zone)]...</span>",
 		"<span class='notice'>[user] begins to saw through the bone in [target]'s [parse_zone(target_zone)].</span>",
 		"<span class='notice'>[user] begins to saw through the bone in [target]'s [parse_zone(target_zone)].</span>")
 
-/datum/surgery_step/saw/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+/datum/surgery_step/saw/tool_check(mob/user, obj/item/tool)
+	if(implement_type == /obj/item && !(tool.get_sharpness() && (tool.force >= 10)))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/organic_steps.dm]: this is why it breaks

`implement_type == /obj/item`

**Review by spookydonut — APPROVED** [review]

**Comment by ATH1909** [comment]
Okay, this is weird. I tested it again with no changes, and now it works... but the chance to succeed when using a sharp item for the saw bone surgery is really, really low (each of these attempts took, like, 25 seconds):
moretesting

**Comment by ExcessiveUseOfCobblestone** [comment]
it's 20% chance of course it's not going to be good.

**Pull request merged**